utils/variant_qc.py

```diff
@@ -0,0 +1,346 @@
+    """
+    Computes lowqual threshold expression for either split or unsplit alleles based on
+    (AS_)QUALapprox
+
+    :param ArrayExpression alleles: Array of alleles
+    :param ArraynumericExpression or NumericExpression qual_approx_expr: QUALapprox or AS_QUALapprox
+    :param int snv_phred_threshold: Phred-scaled SNV "emission" threshold (similar to GATK emission threshold)
+    :param int snv_phred_het_prior: Phred-scaled SNV heterozygosity prior (30 = 1/1000 bases, GATK default)
+    :param int indel_phred_threshold: Phred-scaled indel "emission" threshold (similar to GATK emission threshold)
+    :param int indel_phred_het_prior: Phred-scaled indel heterozygosity prior (30 = 1/1000 bases, GATK default)
+    :return: lowqual expression (BooleanExpression if `qual_approx_expr`is Numeric, Array[BooleanExpression] if `qual_approx_expr` is ArrayNumeric)
+    :rtype: BooleanExpression or ArrayExpression
+    """
+    def low_qual_expr(ref: hl.expr.StringExpression, alt: hl.expr.StringExpression, qual_approx: hl.expr.NumericExpression) -> BooleanExpression:
+        return hl.cond(
+            hl.is_snp(ref, alt),
+            qual_approx < snv_phred_threshold + snv_phred_het_prior,
+            qual_approx < indel_phred_threshold + indel_phred_het_prior
+        )
+    if isinstance(qual_approx_expr, hl.expr.ArrayNumericExpression):
+        return hl.range(1, hl.len(alleles)).map(lambda ai: low_qual_expr(alleles[0], alleles[ai], qual_approx_expr[ai - 1]))
+    else:
+        return low_qual_expr(alleles[0], alleles[1], qual_approx_expr)
+
+
+def generate_fam_stats_expr(
+        trio_mt: hl.MatrixTable,
+        transmitted_strata: Dict[str, Optional[hl.expr.BooleanExpression]] = {'raw': None},
+        de_novo_strata: Dict[str, Optional[hl.expr.BooleanExpression]] = {'raw': None},
+        proband_is_female_expr: Optional[hl.expr.BooleanExpression] =  None
+) -> hl.expr.StructExpression:
+    """
+    Generates a row-wise expression containing the following counts:
+    - Number of alleles in het parents transmitted to the proband
+    - Number of alleles in het parents not transmitted to the proband
+    - Number of de novo mutations
+
+    Both transmission and de novo mutation metrics can be stratified using additional filters.
+    If an empty dict is passed as one of the strata arguments, then this metric isn't computed.
+
+    :param MatrixTable trio_mt: A trio stndard trio MT (with the format as produced by hail.methods.trio_matrix
+    :param dict of str -> BooleanExpression transmitted_strata: Strata for the transmission counts
+    :param dict of str -> BooleanExpression de_novo_strata: Strata for the de novo counts
+    :param BooleanExpression proband_is_female_expr: An optional expression giving the sex the proband. If not given, DNMs are only computed for autosomes.
+    :return: An expression with the counts
+    :rtype: StructExpression
+    """
+
+    # Create map for transmitted, untransmitted and DNM
```

Comment:
Are there any other cases where having these values would be useful? If so maybe move outside this function

Reply:
I'm not sure.. I thought about this for a while and I think maybe it's best to keep it there for now and take it out if needed? But happy to take it out too if you want.

Reply:
Sounds good. If there is nothing else that will use it then it makes sense to keep here